src/main/Core/ImageGenerator/Linux/LinuxAppIconExtractor.ts

```diff
@@ -273,15 +275,29 @@ export class LinuxAppIconExtractor implements FileIconExtractor {
     }
 
     private async getIconThemeName(): Promise<string> {
-        const currentDesktopEnvironment = this.environmentVariablePRovider.get("XDG_SESSION_DESKTOP");
-
-        const iconThemeNameMap: Record<"cinnamon" | "gnome", () => Promise<string>> = {
-            cinnamon: () =>
+        const iconThemeNameMap: Partial<Record<LinuxDesktopEnvironment, () => Promise<string>>> = {
+            GNOME: () => this.commandlineUtility.executeCommand("gsettings get org.gnome.desktop.interface icon-theme"),
+            KDE: () =>
+                this.commandlineUtility.executeCommand("kreadconfig5 --file kdeglobals --group Icons --key Theme"),
+            MATE: () => this.commandlineUtility.executeCommand("gsettings get org.mate.interface icon-theme"),
+            XFCE: () => this.commandlineUtility.executeCommand("gsettings get org.gnome.desktop.interface icon-theme"),
+            Cinnamon: () =>
                 this.commandlineUtility.executeCommand("gsettings get org.cinnamon.desktop.interface icon-theme"),
-            gnome: () => this.commandlineUtility.executeCommand("gsettings get org.gnome.desktop.interface icon-theme"),
+            Pantheon: () =>
+                this.commandlineUtility.executeCommand("gsettings get org.gnome.desktop.interface icon-theme"),
         };
 
-        return (await iconThemeNameMap[currentDesktopEnvironment]()).trim().slice(1, -1);
+        if (!iconThemeNameMap[this.linuxDesktopEnvironmentResolver.resolve()]) {
+            throw new Error(
+                `No available function to read icon theme for this desktop environment. Desktop: ${this.linuxDesktopEnvironmentResolver.resolve()}`,
+            );
+        }
```

Comment:
In my opinion we should not even instantiate this class if the current desktop environment is not one of those defined here. Then we don't need this error handling here. What do you think?